### ISSUE TITLE
Bug: synchronizationStatusUpdated Callback Stops Working Until Client Reconnection

### ISSUE THREAD

**Issue by kuchmiyalex** [twilio/conversations-ios#59]
### SDK version

4.0.3

### Device model

iPhone 14 Pro Max 

### OS version

17.4

## Title: iOS App Not Receiving Synchronization Status Updates for Twilio Conversations

**Description:**

### Issue Overview
After launching the iOS app and establishing connection with TwilioClient, the app correctly receives synchronization status updates for conversations when in the foreground. However, upon transitioning to the background and making changes in conversations, the app fails to receive synchronization status updates for some conversations upon returning to the foreground. This occurs even for conversations in which the user is actively participating.

### Reproduction Steps
1. Launch the iOS app.
2. Connect to TwilioClient.
3. Transition the app to the background.
4. Make changes in conversations (e.g., adding or removing conversations).
5. Bring the app back to the foreground.
6. Observe that some conversations do not receive synchronization status updates.

### Expected Behavior
The app should consistently receive synchronization status updates for all conversations, even after transitioning between the background and foreground.

### Current Behavior
After returning to the foreground, some conversations fail to trigger the synchronizationStatusUpdated callback, resulting in missing updates for those conversations. During this time, it is not possible to fetch messages or send new messages within the affected conversations.

### Additional Information
- Reconnecting the client using `TwilioConversationsClient.conversationsClient(withToken: ...)` resolves the issue temporarily, even with the same token.
- Once reconnection is performed, all conversation updates are received as expected.
- This issue specifically affects users with 500 or more conversations.

**Comment by kuchmiyalex** [comment]
@berkus hey is that something that you might check? Do you need any additional debug info which I can provide? 
At this point, the solution was to decrease the number of active conversations the user participates in, to around 100-150. That solved the problem in 90% cases.

**Comment by berkus** [comment]
@kuchmiyalex not really, I'm not in Twilio anymore, but @AlejandroOrozco should be able to assist.
